### PR TITLE
Add update_value private function to lv_arc to prevent set_value return checks in the event of…

### PR DESCRIPTION
**Pull request by acumartini**
… a range change

### CODE REVIEW

**Comment by acumartini** [comment]
@kisvegabor Also removed all mentions of `USE_ROTARY` left over from the POC work.

**Comment by acumartini** [comment]
Simple example verified working:
```
    lv_obj_t *rotary = lv_arc_create(pg, NULL);
    lv_obj_set_size(rotary, 90, 90);
    lv_obj_align(rotary, pg, LV_ALIGN_IN_TOP_LEFT, 50, 50);
    lv_arc_set_range(rotary, -20000, 20000);
    lv_arc_set_value(rotary, 0);
    lv_arc_set_chg_rate(rotary, 500);
    lv_arc_set_adjustable(rotary, true);
```

**Comment by kisvegabor** [comment]
Hi,

Instead of adding a new "member variable" wouldn't it  be cleaner to create an internal `value_update(arc)` function which is called in `lv_arc_set_value()` after "change check" and inthe end of `lv_arc_set_range`?

**Comment by acumartini** [comment]
Good call. I was on the fence with that member variable. Will make the change.

**Comment by acumartini** [comment]
OK. CR addressed. Return checks performed in set_range are still range specific, and in set_value are value specific. Both call the value_update function if the return checks are passed.

File: src/lv_widgets/lv_arc.c [hunk 1]
```diff
@@ -379,23 +380,13 @@ void lv_arc_set_type(lv_obj_t * arc, lv_arc_type_t type)
 }
 
 /**
- * Set a new value on the arc
+ * Used internally to update arc angles after a value change
  * @param arc pointer to a arc object
  * @param value new value
  */
-void lv_arc_set_value(lv_obj_t * arc, int16_t value)
+void value_update(lv_obj_t * arc)
```
Comment (kisvegabor) [inline src/lv_widgets/lv_arc.c]: As it's an internal function please make it `static` and move to the bottom of the file.

**Review by kisvegabor — COMMENTED** [review]
Looks good thank you for the update.
I found only a minor thing. (See the comment)

File: src/lv_widgets/lv_arc.c (hunk 1 above)
Comment (acumartini) [reply src/lv_widgets/lv_arc.c]: will do

Comment (acumartini) [reply src/lv_widgets/lv_arc.c]: moved to header and made it inline static. Technically not a "private" function anymore, but for the sake of consistency I get why you wanted the change.

Comment (embeddedt) [reply src/lv_widgets/lv_arc.c]: I think @kisvegabor wanted it `static` (not inline) in the C file.

Comment (acumartini) [reply src/lv_widgets/lv_arc.c]: ah gotcha, will do

Comment (acumartini) [reply src/lv_widgets/lv_arc.c]: done @embeddedt ty for clarification

**Comment by kisvegabor** [comment]
Looks good, thank you very much!

As it's rather a bugfix I merge it to master.

**Comment by kisvegabor** [comment]
For simplicity, I need to merge it to dev as the PR affects `dev` features like arc `adjustable`.

**Comment by acumartini** [comment]
Yeah I originally wanted to merge to master but realized I had pulled in the dev changes. Sry to delay the bugfix on master.

**Comment by kisvegabor** [comment]
No problem, next Tuesday it will be merged to master. :slightly_smiling_face:

**Pull request merged**